File: SYP.Web/Modules/Email/EmailAttachments/EmailAttachmentsForm.cs
namespace SYP.Email.Forms;

[FormScript("Email.EmailAttachments")]
[BasedOnRow(typeof(EmailAttachmentsRow), CheckNames = true)]
public class EmailAttachmentsForm
{
    [ReadOnly(true)]
    public long EmailQueueId { get; set; }

    [ReadOnly(true)]
    public string FileName { get; set; }

    [ReadOnly(true)]
    public string ContentType { get; set; }

    [ReadOnly(true)]
    public string FilePath { get; set; }

    [ReadOnly(true)]
    public long FileSize { get; set; }

    [ReadOnly(true)]
    public bool IsInline { get; set; }

    [ReadOnly(true)]
    public string ContentId { get; set; }

    [ReadOnly(true)]
    public DateTime InsertDate { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Email/EmailAttachmentsForm.ts
import { BooleanEditor, DateEditor, initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface EmailAttachmentsForm {
    EmailQueueId: StringEditor;
    FileName: StringEditor;
    ContentType: StringEditor;
    FilePath: StringEditor;
    FileSize: StringEditor;
    IsInline: BooleanEditor;
    ContentId: StringEditor;
    InsertDate: DateEditor;
}

export class EmailAttachmentsForm extends PrefixedContext {
    static readonly formKey = 'Email.EmailAttachments';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmailAttachmentsForm.init) {
            EmailAttachmentsForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;
            var w2 = DateEditor;

            initFormType(EmailAttachmentsForm, [
                'EmailQueueId', w0,
                'FileName', w0,
                'ContentType', w0,
                'FilePath', w0,
                'FileSize', w0,
                'IsInline', w1,
                'ContentId', w0,
                'InsertDate', w2
            ]);
        }
    }
}